src/analysis/static/graph/CFG.ts

```diff
@@ -50,4 +51,70 @@ export class CFG {
       .filter((node) => node.type === NodeType.Root)
       .map((node) => <RootNode>node);
   }
+
+  getNodeById(nodeId : string): Node {
+    const found = this._nodes.filter((node: Node) => node.id == nodeId);
+    if(found.length != 1) {
+      console.log("No node with such id in CFG");
+      return null;
+    }
+    return found[0];
+  }
+
+  getRotatedAdjacencyList(): Map<String, Pair<string, number>[]> {
+    let adjList = new Map<String, Pair<string, number>[]>();
+
+    for(const edge of this._edges){
+      if(!adjList.has(edge.from)){
+        adjList.set(edge.from, []);
+      } 
+      if(!adjList.has(edge.to)){
+        adjList.set(edge.to, []);
+      }
+      adjList.get(edge.to).push(
+        {
+          "first" : edge.from, 
+          "second" : edge.branchType !== undefined ? 1 : 0
+        }
+      );
+    }
+
+    return adjList;
+  }
+
+  findClosestAncestor(from: string, targets: string[]): [number, Node] {
```

Comment:
I would suggest to change targets type to Set<string> immediately instead of converting inside the method.